Mathlib/MeasureTheory/Measure/ProbabilityMeasure.lean

```diff
@@ -166,7 +166,6 @@ theorem toMeasure_comp_toFiniteMeasure_eq_toMeasure (ν : ProbabilityMeasure Ω)
   rfl
 #align measure_theory.probability_measure.coe_comp_to_finite_measure_eq_coe MeasureTheory.ProbabilityMeasure.toMeasure_comp_toFiniteMeasure_eq_toMeasure
 
-@[simp]
 theorem coeFn_comp_toFiniteMeasure_eq_coeFn (ν : ProbabilityMeasure Ω) :
     (ν.toFiniteMeasure : Set Ω → ℝ≥0) = (ν : Set Ω → ℝ≥0) :=
   rfl
```

Comment:
This one makes me sad. Can it at least be `norm_cast`?

Reply:
Is there somewhere this would be helpful?

Reply:
I'm confused why you would be sad about this, unless you have particular knowledge of it being needed as a simp lemma in a downstream project.

I guess I will just add back `norm_cast` to avoid this PR getting blocked up on this. I want this in asap, as it makes life drastically better when working in files with `import Mathlib`.

Reply:
To be clear, here is the type:
```
theorem MeasureTheory.ProbabilityMeasure.coeFn_comp_toFiniteMeasure_eq_coeFn.{u_1} : ∀ {Ω : Type u_1}
  [inst : MeasurableSpace Ω] (ν : ProbabilityMeasure Ω),
  (fun s ↦ (↑ν.toFiniteMeasure s).toNNReal) = fun s ↦ (↑ν s).toNNReal :=
fun {Ω} [MeasurableSpace Ω] ν ↦ rfl
```
i.e. both LHS and RHS are a lambda, meaning that the DiscrTree can't doing anything with it and it is applied everywhere. I don't think I want it in norm_cast either.

Reply:
I am not sure I understand why this is a bad simp lemma, can you elaborate a little bit (maybe adding a comment explaining why it is bad just before the lemma)? (I mean, why wouldn't it only try to apply it to `ProbabilityMeasure`s?)

As a replacement, can we have `ν.toFiniteMeasure s = ν s` for a given set `s`, as a simp lemma, or will it have the same issue?

Reply:
Is this a sign that the `Coe` instance should be defined differently?

Reply:
@sgouezel I've added a comment. That replacement works fine, so I've changed it.

If you'd like to change the name to reflect the change in statement, please either push directly to the PR, or lets do it in a followup.

Reply:
Couldn't the DiscrTree index the type in the lambdas (here NNReal), which means it wouldn't be tried always? Anyway, this is off topic for this PR.

Reply:
The current DiscrTree doesn't do this. Jovan has an alternative DiscrTree that does, but it is unlikely to land in Lean anytime soon.

Reply:
Thanks for update, @sgouezel. Looks good to me now!

Reply:
I've opened #13171